src/modules/BootSupportCheck.rb

```diff
@@ -101,12 +104,11 @@ def correct_loader_type(lt)
     def check_gpt_reserved_partition
       return true unless stage1.mbr?
 
-      devices = Storage.GetTargetMap
-      mbr_disk = Storage.GetDisk(devices, BootStorage.mbr_disk)
-      boot_device = Storage.GetPartition(devices, BootStorage.BootPartitionDevice)
-      return true if mbr_disk["label"] != "gpt"
-      return true if boot_device["used_fs"] != :btrfs
-      return true if mbr_disk["partitions"].any? { |p| p["fsid"] == Partitions.fsid_bios_grub }
+      disk = staging.disks.with(name: BootStorage.mbr_disk).first
+      boot_device = staging.disks.partitions.with(name: BootStorage.BootPartitionDevice).first
```

Comment:
NP: you can do `staging.partitions` directly here. If you are not filtering by disk there is no need to pass through that collection.